src/common/fixNames.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * This pass does the following:
+ *
+ * 1. Introduces declarations for any free identifiers in a program.
+ *    Some older benchmarks, such as SunSpider, have free identifiers and this
+ *    helps us tackle them without modification.
+ * 2. Renames identifiers that Stopify uses for its instrumentation. This is
+ *    necessary (but not sufficient) to ensure that the program cannot tamper
+ *    with the instrumentation.
+ * 3. Leaves references to Stopify-provided functions (e.g., callCC) free.
+ */
+import * as babel from 'babel-core';
+import { NodePath, Visitor } from 'babel-traverse';
+import * as t from 'babel-types';
+import * as fs from 'fs';
+
+// These are identifiers that the rest of Stopify will define and may
+// appear free in the source program.
+const available = [ "callCC" ];
+// These are identifiers that Stopify uses in its instrumentation. If they
+// appear in the original source program, they need to be renamed.
+const internal = [ "target", "stopify" ];
+
+const reserved = [...available, ...internal];
+
+// We use BindingIdentifier and ReferencedIdentifier in this visitor,
+// but the are not part of the TypeScript type definition. The problem
+// with using just Identifier is that it also visits non-computed
+// field lookup expressions.
+const visitIdentifiers = {
+  // BindingIdentifier is a bit of a misnomer. This method visits
+  // identifiers in binding position and in L-value position.
+  BindingIdentifier(path: NodePath<t.Identifier>) {
+    if (!path.scope.hasBinding(path.node.name)) {
+      this.found.add(path.node.name);
+    }
+  },
+  ReferencedIdentifier(path: NodePath<t.Identifier>) {
+    if (available.includes(path.node.name)) {
+      // This may be a legit reference to an identifier provided by Stopify.
+      return;
+    }
+    if (!path.scope.hasBinding(path.node.name)) {
+      this.found.add(path.node.name);
+    }
+  }
+}
+
+function handleScope(path: NodePath<t.Scopable>) {
+  const shadows = Object.keys(path.scope.bindings)
+    .filter(x => reserved.includes(x));
+  for (const x of shadows) {
+    path.scope.rename(x);
+  }
+}
+
+const visitor: Visitor = {
+  Program(path: NodePath<t.Program>) {
+    const found = new Set();
+    path.traverse(visitIdentifiers, { found });
```

Comment:
For future reference:

The pattern to follow to avoid `path.traverse` calls in a `Program` visitor is to make the `Program` visit on exit like so:
```
const visitor: Visitor = {
  Program: {
    exit(path: NodePath<t.Program>): void {
      ....
    }
  },
  ...
}
```
This way you can pull your nested visitor into the same visitor as your `Program` one, the other nodes can accumulate global state, and the `Program` node happens after everything else is done. You can use
```
const visitor: Visitor = {
  Program: {
    enter(path: NodePath<t.Program>): void {
      ...
    },
    exit(...) ...
  },
  ...
}
```
If you need to do any global state initialization using the `Program`'s `path.scope` to do things like get all bound identifiers, etc, on entering the program node.